### PR TITLE
Change extension to be a soft keyword again

### PR DESCRIPTION
**Pull request by tgodzik**
Previously, when discussing https://github.com/lampepfl/dotty/issues/10076 it was mentioned that `extension` would become
a keyword, but turns out it will not be. I kind of jumped the gun here and we should not have chanegd it until it was in
the compiler code.

### CODE REVIEW

**Review by kpbochenek — CHANGES_REQUESTED** [review]
not happy we have to go back to it being softKW, but at least it sounds like final decision.

Good job monitoring those changes in compiler :+1:

File: tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/ExtensionMethodsSuite.scala [hunk 1]
```diff
@@ -59,7 +59,7 @@ class ExtensionMethodsSuite extends BaseDottySuite {
   }
 
   test("modifier-method-indent") {
-    val code = """|extension (c: Circle):
+    val code = """|extension (c: Circle)
```
[the remark this answers is not in the document]
Comment (tgodzik) [reply tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/ExtensionMethodsSuite.scala:62]: It's not really optional :/

```
[error] -- Error: /home/tgodzik/Documents/workspaces/dotty-vinz/src/main/scala/Main.scala:10:23 
[error] 10 |    extension (a : Int):
[error]    |                       ^
[error]    |                       no `:` expected here
```

File: tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/ExtensionMethodsSuite.scala [hunk 2]
```diff
@@ -196,6 +197,18 @@ class ExtensionMethodsSuite extends BaseDottySuite {
...
+  test("extension-named-method") {
+    runTestError[Stat](
+      """|object A{
+         |  def extension(a : Int) = a + 2
+         |  extension(2)
+         |}""".stripMargin,
+      "identifier expected but integer constant found"
+    )
+  }
+
```
[the remark this answers is not in the document]
Comment (tgodzik) [reply tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/ExtensionMethodsSuite.scala:267]: Good catch, it was not actually working. Thanks!

File: tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/ExtensionMethodsSuite.scala [hunk 3]
```diff
@@ -196,6 +197,18 @@ class ExtensionMethodsSuite extends BaseDottySuite {
     )
   }
 
+  // Scala 3 doesn't allow for methods named `extension` to be invoked
+  // https://github.com/lampepfl/dotty/issues/10076
+  test("extension-named-method") {
+    runTestError[Stat](
```
[the remark this answers is not in the document]
Comment (tgodzik) [reply tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/ExtensionMethodsSuite.scala:203]: Added and after some rework it should be fine now.

File: scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala [hunk 4]
```diff
@@ -33,7 +33,7 @@ import scala.util.Failure
 
 class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
   require(Set("", EOL).contains(dialect.toplevelSeparator))
-  private val soft = new SoftKeywords(dialect)
+  private val soft = new SoftKeywords(dialect, (token: Token) => token.next)
```
Comment (tgodzik) [inline scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala:36]: next is needed to make sure that `[` or `(` is actually next. Might be problematic to use it in scalafmt, but I think doable? @olafurpg

**Review by kpbochenek — APPROVED** [review]
LGTM :+1:

File: scalameta/trees/shared/src/main/scala/scala/meta/internal/prettyprinters/TreeSyntax.scala [hunk 5]
```diff
@@ -266,11 +266,17 @@ object TreeSyntax {
       def isAmbiguousWithPatVarType(t: Type.Name, p: Tree): Boolean = {
```
[the remark this answers is not in the document]
Comment (tgodzik) [reply scalameta/trees/shared/src/main/scala/scala/meta/internal/prettyprinters/TreeSyntax.scala]: Yeah, I was kind of afraid it's some magic. Removed it!

**Pull request merged**